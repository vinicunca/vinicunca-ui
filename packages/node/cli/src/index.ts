import { program } from 'commander';
import { errorAndExit } from '@vinicunca/node-utils';
import { build } from '@vinicunca/build';

import { version } from '../package.json';

program.name('Vinicunca CLI').version(version);

program
  .command('build [packageName]')
  .description('build package')
  .action((name: string) => build(name).catch((err) => errorAndExit(err)));

program.parse(process.argv);

export * from '@vinicunca/build';
