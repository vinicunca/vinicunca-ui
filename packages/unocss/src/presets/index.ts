export { PresetVinicuncaOptions, presetVinicunca } from './core';
export { presetWebFonts, WebFontsOptions } from './web-fonts';
