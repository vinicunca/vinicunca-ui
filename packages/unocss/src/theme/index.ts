export * from './border';
export * from './color';
export * from './default';
export * from './font';
export * from './layout';
export * from './shadow';
export * from './size';
export * from './preflight';
export * from './transition';
export * from './types';
