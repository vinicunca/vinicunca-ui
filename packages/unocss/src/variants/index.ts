export * from './breakpoints';
export * from './combinators';
export * from './dark';
export * from './default';
export * from './directions';
export * from './media';
export * from './misc';
export * from './placeholder';
export * from './pseudo';
export * from './important';
export * from './negative';
