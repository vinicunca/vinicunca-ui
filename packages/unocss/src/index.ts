import type { UserConfig } from '@unocss/core';
import type { Theme } from './theme';
import type { WebFontsOptions } from '@vinicunca/preset-web-fonts';
import type { PresetVinicuncaOptions } from './presets';

import presetIcons from '@unocss/preset-icons';
import transformerVariantGroup from '@unocss/transformer-variant-group';
import transformerDirectives from '@unocss/transformer-directives';
import { presetWebFonts } from '@vinicunca/preset-web-fonts';

import { presetVinicunca } from './presets';

export function extendUnocssOptions(
  { config = {}, vinicunaConfig = {}, webFontsConfig }:
  { config?: UserConfig; vinicunaConfig?: PresetVinicuncaOptions; webFontsConfig?: WebFontsOptions } = {},
): UserConfig<Theme> {
  const include = [
    /\.vue$/,
    /vinicunca\/es\/components/,
    /\.ts$/,
    /\.tsx$/,
  ];

  if (config.include) {
    if (Array.isArray(config.include)) {
      include.push(...config.include);
    } else {
      include.push((config.include as any));
    }
  }

  const presets = [
    presetVinicunca(vinicunaConfig),
    presetIcons({
      scale: 1.2,
    }),
  ];

  if (webFontsConfig) {
    presets.push(presetWebFonts(webFontsConfig));
  }

  return {
    ...config,

    presets,

    transformers: [
      transformerDirectives(),
      transformerVariantGroup(),
    ],

    include,
  };
}

