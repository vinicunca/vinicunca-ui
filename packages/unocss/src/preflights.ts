import type { Preflight, PreflightContext } from '@unocss/core';
import type { Theme } from './theme';
import type { PresetVinicuncaOptions } from './presets';

import { entriesToCss } from '@unocss/core';

export function preflights(options: PresetVinicuncaOptions = {}): Preflight[] {
  return [
    {
      layer: 'preflights',
      getCSS(ctx: PreflightContext<Theme>) {
        if (ctx.theme.preflightBase) {
          const _brands: any = options.brands || {};

          const brandVars = Object.keys(_brands).reduce<any>((prev, acc) => {
            const _key = `--vin-brand-${acc}`;
            prev[_key] = _brands[acc];
            return prev;
          }, {});

          const variables = {
            ...brandVars,
            ...ctx.theme.varsBase,
          };

          const rootCss = entriesToCss(Object.entries(variables));
          const root = `:root{${rootCss}}`;

          const globalCss = entriesToCss(Object.entries(ctx.theme.preflightBase));
          const global = `*,::before,::after{${globalCss}}::backdrop{${globalCss}}`;

          return `${root}; ${global}`;
        }
      },
    },
  ];
}
