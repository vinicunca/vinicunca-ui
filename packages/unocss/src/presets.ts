import type { Postprocessor, Preset, PresetOptions } from '@unocss/core';
import type { Theme, ThemeAnimation } from './theme';

import { preflights } from './preflights';
import { rules } from './rules';
import { theme } from './theme';
import { variants } from './variants';

export type { ThemeAnimation, Theme };

const brands = ['primary', 'secondary', 'success', 'info', 'warning', 'danger'];

export interface PresetVinicuncaOptions extends PresetOptions {
  /**
   * @default 'class'
   */
  dark?: 'class' | 'media';
  /**
   * @default false
   */
  attributifyPseudo?: Boolean;
  /**
   * @default 'vin-'
   */
  variablePrefix?: string;

  brands?: {
    primary?: string;
    secondary?: string;
    success?: string;
    info?: string;
    warning?: string;
    danger?: string;
  };
}

export function presetVinicunca(options: PresetVinicuncaOptions = {}): Preset<Theme> {
  options.dark = options.dark ?? 'class';
  options.attributifyPseudo = options.attributifyPseudo ?? false;

  return {
    name: '@vinicunca/unocss-preset',
    theme,
    rules,
    variants: variants(options),
    options,
    postprocess: options.variablePrefix && options.variablePrefix !== 'un-'
      ? VarPrefixPostprocessor(options.variablePrefix)
      : undefined,
    preflights: preflights(options),
    safelist: [
      ...brands.map((c) => `bg-${c}`),
      ...brands.map((c) => `hover:bg-${c}`),
      ...brands.map((c) => `border-${c}`),
      ...brands.map((c) => `text-${c}`),
      ...brands.map((c) => `shadow-${c}`),
    ],
  };
}

function VarPrefixPostprocessor(prefix: string): Postprocessor {
  return (obj) => {
    obj.entries.forEach((i) => {
      i[0] = i[0].replace(/^--vin-/, `--${prefix}`);
      if (typeof i[1] === 'string') {
        i[1] = i[1].replace(/var\(--vin-/g, `var(--${prefix}`);
      }
    });
  };
}
