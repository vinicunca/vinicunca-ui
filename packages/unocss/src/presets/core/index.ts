import type { Postprocessor, Preset, PresetOptions } from 'unocss';
import type { Theme, ThemeAnimation } from '../../theme';

import { preflights } from '../../preflights';
import { rules } from '../../rules';
import { theme } from '../../theme';
import { variants } from '../../variants';

export type { ThemeAnimation, Theme };

export interface PresetVinicuncaOptions extends PresetOptions {
  /**
   * @default 'class'
   */
  dark?: 'class' | 'media';
  /**
   * @default false
   */
  attributifyPseudo?: Boolean;
  /**
   * @default 'vin-'
   */
  variablePrefix?: string;
}

export function presetVinicunca(options: PresetVinicuncaOptions = {}): Preset<Theme> {
  options.dark = options.dark ?? 'class';
  options.attributifyPseudo = options.attributifyPseudo ?? false;

  return {
    name: '@vinicunca/unocss-preset',
    theme,
    rules,
    variants: variants(options),
    options,
    postprocess: options.variablePrefix && options.variablePrefix !== 'un-'
      ? VarPrefixPostprocessor(options.variablePrefix)
      : undefined,
    preflights: preflights(options),
    safelist: [
      // Used by useLockScreen
      'overflow-hidden',
    ],
  };
}

function VarPrefixPostprocessor(prefix: string): Postprocessor {
  return (obj) => {
    obj.entries.forEach((i) => {
      i[0] = i[0].replace(/^--vin-/, `--${prefix}`);
      if (typeof i[1] === 'string') {
        i[1] = i[1].replace(/var\(--vin-/g, `var(--${prefix}`);
      }
    });
  };
}
