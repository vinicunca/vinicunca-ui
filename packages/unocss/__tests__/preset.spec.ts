import { createGenerator } from '@unocss/core';
import { describe, expect, test } from 'vitest';

import { presetVinicunca } from '../src/presets';

const uno = createGenerator({
  presets: [
    presetVinicunca({
      brands: {
        primary: '#1976d2',
        secondary: '#9c27b0',
        success: '#2e7d32',
        info: '#0288d1',
        warning: '#ed6c02',
        danger: '#d32f2f',
      },
    }),
  ],
});

describe('preset-mini', () => {
  test('preflight', async () => {
    const { css } = await uno.generate('');
    expect(css).toMatchSnapshot();
  });
});
