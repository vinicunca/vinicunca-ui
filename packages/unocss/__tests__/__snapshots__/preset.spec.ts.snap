// Vitest Snapshot v1

exports[`preset-mini > preflight 1`] = `
"/* layer: preflights */
:root{--vin-brand-primary:#1976d2;--vin-brand-secondary:#9c27b0;--vin-brand-success:#2e7d32;--vin-brand-info:#0288d1;--vin-brand-warning:#ed6c02;--vin-brand-danger:#d32f2f;--vin-overlay-color-lighter:rgba(0, 0, 0, 0.5);}; *,::before,::after{--vin-rotate:0;--vin-rotate-x:0;--vin-rotate-y:0;--vin-rotate-z:0;--vin-scale-x:1;--vin-scale-y:1;--vin-scale-z:1;--vin-skew-x:0;--vin-skew-y:0;--vin-translate-x:0;--vin-translate-y:0;--vin-translate-z:0;--vin-ring-offset-shadow:0 0 rgba(0,0,0,0);--vin-ring-shadow:0 0 rgba(0,0,0,0);--vin-shadow-inset: ;--vin-shadow:0 0 rgba(0,0,0,0);--vin-ring-inset: ;--vin-ring-offset-width:0px;--vin-ring-offset-color:#fff;--vin-ring-width:0px;--vin-ring-color:rgba(147,197,253,0.5);--vin-blur: ;--vin-brightness: ;--vin-contrast: ;--vin-drop-shadow: ;--vin-grayscale: ;--vin-hue-rotate: ;--vin-invert: ;--vin-saturate: ;--vin-sepia: ;--vin-backdrop-blur: ;--vin-backdrop-brightness: ;--vin-backdrop-contrast: ;--vin-backdrop-grayscale: ;--vin-backdrop-hue-rotate: ;--vin-backdrop-invert: ;--vin-backdrop-opacity: ;--vin-backdrop-saturate: ;--vin-backdrop-sepia: ;}::backdrop{--vin-rotate:0;--vin-rotate-x:0;--vin-rotate-y:0;--vin-rotate-z:0;--vin-scale-x:1;--vin-scale-y:1;--vin-scale-z:1;--vin-skew-x:0;--vin-skew-y:0;--vin-translate-x:0;--vin-translate-y:0;--vin-translate-z:0;--vin-ring-offset-shadow:0 0 rgba(0,0,0,0);--vin-ring-shadow:0 0 rgba(0,0,0,0);--vin-shadow-inset: ;--vin-shadow:0 0 rgba(0,0,0,0);--vin-ring-inset: ;--vin-ring-offset-width:0px;--vin-ring-offset-color:#fff;--vin-ring-width:0px;--vin-ring-color:rgba(147,197,253,0.5);--vin-blur: ;--vin-brightness: ;--vin-contrast: ;--vin-drop-shadow: ;--vin-grayscale: ;--vin-hue-rotate: ;--vin-invert: ;--vin-saturate: ;--vin-sepia: ;--vin-backdrop-blur: ;--vin-backdrop-brightness: ;--vin-backdrop-contrast: ;--vin-backdrop-grayscale: ;--vin-backdrop-hue-rotate: ;--vin-backdrop-invert: ;--vin-backdrop-opacity: ;--vin-backdrop-saturate: ;--vin-backdrop-sepia: ;}
/* layer: default */
.border-danger{border-color:var(--vin-brand-danger);}
.border-info{border-color:var(--vin-brand-info);}
.border-primary{border-color:var(--vin-brand-primary);}
.border-secondary{border-color:var(--vin-brand-secondary);}
.border-success{border-color:var(--vin-brand-success);}
.border-warning{border-color:var(--vin-brand-warning);}
.bg-danger,
.hover\\\\:bg-danger:hover{background-color:var(--vin-brand-danger);}
.bg-info,
.hover\\\\:bg-info:hover{background-color:var(--vin-brand-info);}
.bg-primary,
.hover\\\\:bg-primary:hover{background-color:var(--vin-brand-primary);}
.bg-secondary,
.hover\\\\:bg-secondary:hover{background-color:var(--vin-brand-secondary);}
.bg-success,
.hover\\\\:bg-success:hover{background-color:var(--vin-brand-success);}
.bg-warning,
.hover\\\\:bg-warning:hover{background-color:var(--vin-brand-warning);}
.text-danger{color:var(--vin-brand-danger);}
.text-info{color:var(--vin-brand-info);}
.text-primary{color:var(--vin-brand-primary);}
.text-secondary{color:var(--vin-brand-secondary);}
.text-success{color:var(--vin-brand-success);}
.text-warning{color:var(--vin-brand-warning);}
.shadow-danger{--vin-shadow-color:var(--vin-brand-danger);}
.shadow-info{--vin-shadow-color:var(--vin-brand-info);}
.shadow-primary{--vin-shadow-color:var(--vin-brand-primary);}
.shadow-secondary{--vin-shadow-color:var(--vin-brand-secondary);}
.shadow-success{--vin-shadow-color:var(--vin-brand-success);}
.shadow-warning{--vin-shadow-color:var(--vin-brand-warning);}"
`;
